share/hooks/ft/host_error.rb

```diff
@@ -283,6 +290,15 @@ def states_xpath(*arr)
             log "delete #{vm_id}"
             vm.delete
         when :migrate
+            vm_ds_id  = vm.retrieve_elements("/VM/HISTORY_RECORDS/HISTORY[position()=last()]/DS_ID")[0]
+
+            ds_xpath  = "/DATASTORE_POOL/DATASTORE[ID=\"#{vm_ds_id}\"]/TEMPLATE/SHARED"
+            is_shared = ds_pool.retrieve_elements(ds_xpath)[0]
```

Comment:
Same here what if no is_shared?

Reply:
Maybe migrate or not migrate by default?

Reply:
> Maybe migrate or not migrate by default?

the idea is to not migrate non shared VMs

Reply:
This should also be safe, however if a datastore somehow doesn't contain SHARED it should skip and it seems this code would allow it to pass.  I'll add a check here as well.